### DISCUSSION TITLE
Date type setting of variables

### DISCUSSION THREAD

**Discussion by abc-hy** [idaholab/moose#22488] [Q&A Modules: Phase field] [answered]
Hi everyone,

Does anyone know how can I set the date type of any variables I define? Currently, I found the data type of the variable I define is **cell data**, however, I want it to be **point data** type, how to set the specific data type for the variables?

Thank you very much,
Best

**Comment by GiudGiud** [comment, marked as answer]
Hello

You set the finite element family with the family parameter in the variables block.
cell_data is usually elemental variables, while point data is nodal variables.

Guillaume

**Reply by abc-hy** [reply]
![image](https://user-images.githubusercontent.com/63679626/197655175-9f3e08c0-e0bc-488d-b9db-edb7de3e412a.png)
This is how I set the variable, how to change it to make this variable become point data? Currently, the data type I got is cell data.

**Reply by GiudGiud** [reply]
You can convert point data to cell data in Paraview directly. There's an option, maybe in the advanced settings after clicking the little cogwheel.

If you want moose to output point data, you ll need to create an auxiliary FIRST (order) LAGRANGE variable and project your CONSTANT MONOMIAL variable

**Reply by abc-hy** [reply]
Got it. Thank you

**Reply by abc-hy** [reply]
![image](https://user-images.githubusercontent.com/63679626/198061213-c5bb4416-85f7-4c8d-84e0-a4890802e888.png)
After I set the laplacian to be FIRST and LAGRANGE, I keep getting this error message. When I change back to set laplacian as CONSTANT and MONOMIAL, I can get it run. Why is this? Does that mean some variable just cannot be set as point data?
Also, last time I used paraview to convert cell data to point data, and I found the data at the interface is very very little, is the convertion able to maintain the accuracy? Or is cell data just not as accurate as point data?

**Reply by GiudGiud** [reply]
Is laplacian a variable?
what is your current [Mesh] block?

**Reply by abc-hy** [reply]
![image](https://user-images.githubusercontent.com/63679626/198123256-dbdc6db0-bdc3-4407-818c-e6f469f3ca67.png)
Laplacian is calculated from w and c, w and c are all variables.
![image](https://user-images.githubusercontent.com/63679626/198123389-610429c7-5ca9-423b-9785-fc744be360bf.png)
This is my mesh

**Reply by GiudGiud** [reply]
nz = 0 is not needed. 

Are you running in parallel?

Do you hit any issue with less refinement?

And there are no second order variables anywhere in the simulation?
